GUI/GUIWindow.cpp

```diff
@@ -2674,3 +2663,136 @@ void UI_Create() {
 
     LoadPartyBuffIcons();
 }
+
+
+String NameAndTitle(const char* name, const char* title) {
+    return localization->FormatString(
+        LSTR_FMT_S_THE_S,
+        name,
+        title
+    );
+}
+
+
+String NameAndTitle(const char* name, PLAYER_CLASS_TYPE class_type) {
+    return NameAndTitle(
+        name,
+        localization->GetClassName(class_type)
+    );
+}
+
+
+String NameAndTitle(const char* name, NPCProf profession) {
+    return NameAndTitle(
+        name,
+        localization->GetNpcProfessionName(profession)
+    );
+}
+
+
+String NameAndTitle(NPCData* npc) {
+    if (npc->pName) {
+        if (npc->profession) {
+            Assert(npc->profession < 59);
+            return NameAndTitle(npc->pName, npc->profession);
+        }
+
+        return npc->pName;
+    }
+
+    return String("");
+}
+
+
+String GetDisplayName(Actor* actor) {
+    if (actor->dword_000334_unique_name)
+        return pMonsterStats->pPlaceStrings[actor->dword_000334_unique_name];
+    else
+        return pMonsterStats->pInfos[actor->pMonsterInfo.uID].pName;
+}
+
+
+static String SeekKnowledgeElswhereString(Player *player) {
+    return localization->FormatString(
+        LSTR_FMT_SEEK_KNOWLEDGE_ELSEWHERE,
+        player->pName,
+        localization->GetClassName(player->classType)
+    )
+    + "\n \n"
+    + localization->GetString(LSTR_NO_FURTHER_OFFERS);
+}
+
+void SeekKnowledgeElswhereDialogueOption(GUIWindow* dialogue, Player* player) {
+    String str = SeekKnowledgeElswhereString(pPlayers[uActiveCharacter]);
+    int text_height = pFontArrus->CalcTextHeight(str, dialogue->uFrameWidth, 0);
+
+    dialogue->DrawTitleText(
+        pFontArrus,
+        0,
+        (174 - text_height) / 2 + 138,
+        Color16(0xFFu, 0xFFu, 0x9Bu),
+        str,
+        3
+    );
+}
+
+
+void SkillTrainingDialogue(
```

Comment:
SkillTrainingDialogue - that's a big chunk i encountered at least 4 times so had to extract it
there are still duplicated fragments around shops logic but i decided to cut it here as PRs got huge already